grr/server/grr_response_server/gui/ui/components/client_details/entry_history_dialog/entry_history_dialog.ts

```diff
@@ -0,0 +1,54 @@
+import {Component, Inject} from '@angular/core';
+import {MAT_DIALOG_DATA} from '@angular/material/dialog';
+import {Client} from '@app/lib/models/client';
+
+export type EntryType =
+    'primitive'|'timestamp'|'size'|'user-list'|'interface-list'|'volume-list';
+
+export interface EntryHistoryDialogParams {
+  path: string;
+  type: EntryType;
+  clientVersions: Client[];
+}
+
+interface EntryHistoryTableRow {
+  time: Date;
+  version: any;
+}
+
+@Component({
+  selector: 'entry-history-dialog',
+  templateUrl: './entry_history_dialog.ng.html',
+  styleUrls: ['./entry_history_dialog.scss'],
+})
+export class EntryHistoryDialog {
+  readonly entryType: EntryType;
+  readonly tableRows: EntryHistoryTableRow[] = [];
+
+  constructor(@Inject(MAT_DIALOG_DATA) private readonly data:
+                  EntryHistoryDialogParams) {
+    if (this.data.path === '') {
+      throw new Error('Empty "path" provided');
+    }
+    this.entryType = data.type;
+    this.initTableRows(this.data);
+  }
+
+  private initTableRows(data: EntryHistoryDialogParams) {
+    data.clientVersions.forEach((client) => {
+      let property: any = client;
+      data.path.split('.').forEach((token) => {
+        property = property[token];
+      });
+
+      if (property === undefined) {
+        throw new Error('Wrong "path" provided');
```

Comment:
This error message is not helpful. Include `data` in the error to make debugging easier.

Reply:
I updated it in the second part: #842